File: 1750-Minimum_Length_of_String_After_Deleting_Similar_Ends/Minimum_Length_of_String_After_Deleting_Similar_Ends.go
package minimumlengthofstringafterdeletingsimilarends

func minimumLength(s string) int {
	left := 0
	right := len(s) - 1

	for left < right {
		lc := s[left]
		rc := s[right]

		if lc != rc {
			return right - left + 1
		}

		for left+1 < right && lc == s[left+1] {
			left++
		}

		for left < right-1 && rc == s[right-1] {
			right--
		}

		right--
		left++
	}

	return right - left + 1
}

File: 1750-Minimum_Length_of_String_After_Deleting_Similar_Ends/Minimum_Length_of_String_After_Deleting_Similar_Ends.java
class Solution {
    public int minimumLength(String s) {
        int left = 0;
        int right = s.length() - 1;
        
        char[] arr = s.toCharArray();
        while(left < right){
            if(arr[left] != arr[right]) {
                return (right - left + 1);
            }
            while(left + 1 < right && arr[left] == arr[left + 1]){
                left++;
            }
            while(left < right - 1 && arr[right] == arr[right-1]){
                right--;
            }
            
            left++;
            right--;
        }
        
        return (right - left + 1);
    }
}